containers/init/clone/clone.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# Copyright 2020 gRPC authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+git clone --recursive $CLONE_REPO /src/workspace
```

Comment:
note that too much cloning from github can actually make the test unstable. At the volume at which we run tests, we actually see git clone failing relatively often (we think it might be because of hitting github's quota). No change needed now, but it's something you should know of because when run at scale, this can actually impact overall stability of the solution.

This is also one of the reasons why working with pre-build docker images can lead to better stability (you can build locally from already checked-out repo or retry more easily and then push ready to use image). As long as we allow both options (build at runtime and build in advance), we are fine.

Reply:
Sure, I think we can also reach out to Prow team if it becomes a bigger issue. They are building at runtime for all the kubernetes repos. I like the immediate start of a prebuilt image, but it also requires some significant cleanup efforts. The design I'm proposing in #5 should make both possible. I have a doc describing it that @paulosjca is reviewing. Once we reach consensus, I can better describe the new design with you :)

Reply:
One other note is that b/c it is part of an init container and not the runtime container itself, we'll be able to easily distinguish if the problem is related to the clone or the test. This can address concerns about logging and progress.